airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py

```diff
@@ -448,13 +448,45 @@ class SellerFeedbackReports(IncrementalReportsAmazonSPStream):
     cursor_field = "Date"
     transformer: TypeTransformer = TypeTransformer(TransformConfig.DefaultSchemaNormalization | TransformConfig.CustomSchemaNormalization)
 
-    @transformer.registerCustomTransform
-    def transform_function(original_value: Any, field_schema: Dict[str, Any]) -> Any:
-        if original_value and "format" in field_schema and field_schema["format"] == "date":
-            transformed_value = pendulum.from_format(original_value, "M/D/YY").to_date_string()
-            return transformed_value
+    region_date_formats = dict(
+        # eu
+        A2VIGQ35RCS4UG = "D/M/YY", # AE
+        A1PA6795UKMFR9 = "D/M/YY", # DE
+        A1C3SOZRARQ6R3 = "D/M/YY", # PL
+        ARBP9OOSHTCHU = "D/M/YY", # EG
+        A1RKKUPIHCS9HS = "D/M/YY", # ES
+        A13V1IB3VIYZZH = "D/M/YY", # FR
+        A21TJRUUN4KGV = "D/M/YY", # IN
+        APJ6JRA9NG5V4 = "D/M/YY", # IT
+        A1805IZSGTT6HS = "D/M/YY", # NL
+        A17E79C6D8DWNP = "D/M/YY", # SA
+        A2NODRKZP88ZB9 = "D/M/YY", # SE
+        A33AVAJ2PDY3EV = "D/M/YY", # TR
+        A1F83G8C2ARO7P = "D/M/YY", # UK
+        # fe
+        A39IBJ37TRP1C6 = "D/M/YY", # AU
+        A1VC38T7YXB528 = "YY/M/D", # JP
+        A19VAU5U5O7RUS = "D/M/YY", # SG
+        # na
+        ATVPDKIKX0DER = "M/D/YY", # US
+        A2Q3Y263D00KWC = "D/M/YY", # BR
+        A2EUQ1WTGCTBG2 = "M/D/YY", # CA
+        A1AM78C64UM0Y8 = "D/M/YY", # MX
+    )
 
-        return original_value
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.transformer.registerCustomTransform(self.get_transform_function())
+
+    def get_transform_function(self):
+        def transform_function(original_value: Any, field_schema: Dict[str, Any]) -> Any:
+            if original_value and "format" in field_schema and field_schema["format"] == "date":
```

Comment:
```suggestion
        super().__init__(*args, **kwargs)

      @transformer.registerCustomTransform
      def transform_function(original_value: Any, field_schema: Dict[str, Any]) -> Any:
            if original_value and "format" in field_schema and field_schema["format"] == "date":
```

Use decorator to register the custom transformation function. Search for `@transformer.registerCustomTransform` in the project to see examples.

Reply:
```python
@transformer.registerCustomTransform
      def transform_function(original_value: Any, field_schema: Dict[str, Any]) -> Any:
```

How would I access `self.marketplace_ids[0]` using decorator approach?